lib/protocol/protocol.go

```diff
@@ -729,7 +729,7 @@ func (c *rawConnection) close(err error) {
 		}
 		c.awaitingMut.Unlock()
 
-		go c.receiver.Close(c.id, err)
```

Comment:
I'm beginning to realize that "go" for these kinds of things as a workaround against locking issues is a quite bad code smell.